extensions/ql-vscode/docs/test-plan.md

```diff
@@ -0,0 +1,257 @@
+```
+"test-repo-list": [
+    "angular-cn/ng-nice",
+    "apache/hadoop",
+    "apache/hive"
+]
+```
+3. Check the notification message. It should:
+   - Show the number of repos that are going to be queried
+   - Provide a link to the actions workflow
+4. Check the query history. It should:
+   - Show that an item has been added to the query history
+   - The item should be marked as “in progress”.
+5. Once the query completes:
+   - Check that a notification is shown with a link to the results view 
+   - Check the results view 
+   - Check the code paths view, including the code paths drop down menu. 
+   - Check that the repository filter box works
+   - Click links to files/locations on GitHub.
+
+### Test Case 2: Running a problem query and viewing results
+
+1. Open the [ReDoS query](https://github.com/github/codeql/blob/main/javascript/ql/src/Performance/ReDoS.ql).
+2. Run it against the "Top 10" repositories.
+3. Check the notification message. It should:
+   - Show the number of repos that are going to be queried
+   - Provide a link to the actions workflow
+4. Check the query history. It should:
+   - Show that an item has been added to the query history
+   - The item should be marked as “in progress”.
+5. Once the query completes:
+   - Check that a notification is shown with a link to the results view
+   - Check that the results are rendered with an alert message and a highlighted code snippet:
+![highlighted-code-snippet](images/highlighted-code-snippet.png)
+
+### Test Case 3: Running a non-problem query and viewing results
+
+1. Open the [FunLinesOfCode query](https://github.com/github/codeql/blob/main/cpp/ql/src/Metrics/Functions/FunLinesOfCode.ql).
+2. Run it against a single repository (e.g. `google/brotli`).
+3. Once the query is complete:
+   - Open the query results
+   - Check that the results show up in a table:
+![results-table](images/results-table.png)
+
+### Test Case 4: Interacting with query history
+
+1. Click a history item (for MRVA):
+   - Check that exporting results works
+   - Check that sorting results works
+   - Check that copying repo lists works 
+2. Open the query directory (containing results):
+   - Check that the correct directory is opened and there are results in it
+3. Open variant analysis on GitHub
+   - Check that the correct workflow is opened
+
+### Test Case 5: Cancelling a variant analysis run
+
+When running one of the above queries, try cancelling one from within VS Code:
+- Check that the query is cancelled and the query history item is updated.
+- Check that the workflow run is also cancelled. 
+- Check that any available results are visible in VS Code.
+
+### Test Case 6: Change to a different colour theme 
+
+When running one of the above queries, try changing to a different colour theme 
+and check that everything looks sensible. Are there any components that are not showing up?
+
+## Optional Test Cases
+
+### 1. Selecting repositories to run on
```

Comment:
Similar comment to further up, these are mostly MRVA test cases so we might want to make that clear. 

I'd also drop the numbering around areas, so I'd go with

```suggestion
### Selecting repositories to run on
```